src/lcm/ndimage.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2019 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Modifications made by Tim Mensinger, 2024
+
+import functools
+import itertools
+import operator
+from collections.abc import Sequence
+
+import jax.numpy as jnp
+from jax import Array, lax, util
+
+
+def map_coordinates(
+    input: Array,
+    coordinates: Sequence[Array],
+) -> Array:
+    """Map the input array to new coordinates using linear interpolation.
+
+    Modified from JAX implementation of :func:`scipy.ndimage.map_coordinates`.
+
+    Given an input array and a set of coordinates, this function returns the
+    interpolated values of the input array at those coordinates. For coordinates outside
+    the input array, linear extrapolation is used.
+
+    Args:
+      input: N-dimensional input array from which values are interpolated.
+      coordinates: length-N sequence of arrays specifying the coordinates
+        at which to evaluate the interpolated values
+
+    Returns:
+      The interpolated (extrapolated) values at the specified coordinates.
+
+    """
+    if len(coordinates) != input.ndim:
```

Comment:
I don't know if this will be optimized away, but the computation is tiny compared to the rest of the function. Also, this condition is JAX-friendly since it does not rely on specific values of the inputs, i.e., the JAX tracers can pass through it.

When experimenting with this function, I got this error multiple times, meaning it's easy to pass the coordinates in the wrong format. Therefore, I believe it's best to keep it.